hamster_gtk/helpers.py

```diff
@@ -177,6 +182,54 @@ def decompose_raw_fact_string(text, raw=False):
     return result
 
 
+# [TODO]
+# Once LIB-251 has been fixed this should no longer be needed.
+def get_recent_activities(controller, start, end):
+    """Return a list of all activities logged in facts within the given timeframe."""
+    # [FIXME]
+    # This manual sorting within python is of cause less than optimal. We stick
+    # with it for now as this is just a preliminary workaround helper anyway and
+    # effective sorting will need to be implemented by the storage backend in
+    # ``hamster-lib``.
+    facts = sorted(controller.facts.get_all(start=start, end=end),
+        key=operator.attrgetter('start'), reverse=True)
+    recent_activities = [fact.activity for fact in facts]
+    return OrderedSet(recent_activities)
+
+
+def serialize_activity(activity, separator='@', none_category='not categorized'):
```

Comment:
We probably do not want to serialize uncategorized activities with the “not categorized” category.

![screenshot from 2018-05-04 14-40-33](https://user-images.githubusercontent.com/705123/39628421-53d9893e-4fa9-11e8-8c16-45e3193de61a.png)

If you copy/start them, the fake category will become real.

Reply:
good catch. :+1: